woocommerce/woocommerce-ios
# [Performance] Only sync orders modified after the last full sync in the order list

Closes: #10042
⚠️ Depends on #10060 ⚠️ 


## Description

When the Orders tab is opened, we want to make a request using the `modified_after` parameter with the last sync date as the value, to only fetch more recent orders (orders created or modified since the last sync). This prevents re-fetching orders that haven't changed.

This PR updates how we perform the sync on the order list:

* If we are performing a full re-sync (on pull to refresh or when applying a new filter) we fetch all orders, regardless of the date they were created or modified.
* If we aren't performing a full re-sync, and we're syncing the first page of data, we set the `modified_after` parameter using the VC's `lastFullSyncTimestamp` as the argument. This fetches any orders created or modified since the last successful sync.
* All other behavior is the same, so once the sync is complete the newly fetched orders are upserted and displayed in the order list as before.

## Testing instructions


1. Build and run the app.
2. Open the Orders tab and confirm your orders list loads.
3. Navigate to another tab.
4. In a web browser, add a new order and/or update an existing order.
5. In the app, open the Orders tab again. (Note that there is a minimum 30-second interval between syncs, so wait until 30 seconds have passed before reopening this tab to ensure it performs a sync.)
6. Confirm the new and updated orders appear as expected in the order list.

You can also test pulling to refresh, scrolling to load to orders, and add/clearing filters to confirm the orders load as expected in the list.

## Screenshots


Note that in this screencast, when the Orders tab is opened a sync is performed. When the sync finishes, a more recent order is added to the top of the order list and the order that was at the top of the list (Order 688) is updated with a new order total (from $90 to $106):

https://github.com/woocommerce/woocommerce-ios/assets/8658164/bd9d51d5-50dd-4ebd-9e83-98fd4a301635


---
